TRENZ.Lib.RazorMail.Core/Models/MailHeaderCollection.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace TRENZ.Lib.RazorMail.Models;
+
+/// <summary>
+/// Represents a collection of mail headers.
+/// </summary>
```

Comment:
this doesn't actually support custom headers yet, does it?

Reply:
Yes it does, it is just a dictionary after all. The mail client implementation just don't use other keys yet. Should be add this in this PR too?

Reply:
Kinda? The keys aren't hardcoded in the implementations though, they use the getters in the collection which in turn have "hardcoded"/known keys.